libs/core/src/lib/token/tokenizer.component.ts

```diff
@@ -204,12 +217,39 @@ export class TokenizerComponent implements AfterViewChecked, AfterViewInit, Afte
             this.tokenListClickSubscriptions.push(token.onTokenClick.subscribe((event) => {
                 event.stopPropagation();
                 this.focusTokenElement(index);
-                this.tokenList.forEach(shadowedToken => {
-                    if (shadowedToken !== token) {
-                        shadowedToken.selected = false;
-                    }
-                });
+                const elementIndexSelected = index;
+                if (!this.ctrlKeyPressed && !this.shiftKeyPressed) {
```

Comment:
Please try to split this code into some functional blocks and try to move them to separate methods. Its hard to understand whats going on here.